File: rb/features.rb
# Arbeitnow SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ArbeitnowFeatures
  def self.make_feature(name)
    case name
    when "base"
      ArbeitnowBaseFeature.new
    when "test"
      ArbeitnowTestFeature.new
    else
      ArbeitnowBaseFeature.new
    end
  end
end

File: py/arbeitnow_sdk/features.py
# Arbeitnow SDK feature factory

from arbeitnow_sdk.feature.base_feature import ArbeitnowBaseFeature
from arbeitnow_sdk.feature.test_feature import ArbeitnowTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArbeitnowBaseFeature(),
        "test": lambda: ArbeitnowTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
